shell/linux/Makefile

```diff
@@ -0,0 +1,212 @@
+
+LOCAL_PATH := $(call my-dir)
+FOR_LINUX :=1
+#NO_REC := 1
+#NO_REND := 1
+WEBUI :=1 
+USE_ALSA := 1
+USE_OSS := 1
+#USE_PULSEAUDIO := 1
+USE_X11 := 1
+
+MFLAGS := 
+ASFLAGS := 
+LDFLAGS :=
+INCS :=
+LIBS :=
+CFLAGS := 
+CXXFLAGS :=
+
+# Platform auto-detection
+# Can be overridden by using:
+#   make platform=AMD64
+ifeq (,$(platform))
+    ARCH = $(shell uname -m)
+    ifeq ($(ARCH), $(filter $(ARCH), i386 i686))
+        platform = i686
+    else ifeq ($(ARCH), $(filter $(ARCH), x86_64 AMD64))
+        platform = AMD64
+    else ifneq (,$(findstring arm,$(ARCH)))
+        HARDWARE = $(shell grep Hardware /proc/cpuinfo)
+        ifneq (,$(findstring BCM2709,$(HARDWARE)))
+            platform = rpi2
+        else ifneq (,$(findstring AM33XX,$(HARDWARE)))
+            platform = beagle
+        else ifneq (,$(findstring Pandora,$(HARDWARE)))
+            platform = pandora
+        else
+            $(error Unsupported Hardware)
+        endif
+    else ifneq (,$(findstring mips,$(ARCH)))
+        platform = gcwz
+    else
+        $(error Unsupported CPU architecture)
+    endif
+endif
+
+$(info Platform: $(platform))
+
+# Generic x86 (32 bit)
+ifneq (,$(findstring i686,$(platform)))
+    X86_REC := 1
+    NOT_ARM := 1
+    MFLAGS += -m32
+    ASFLAGS += -32
+    LDFLAGS += -m32
+    CFLAGS += -m32 -D TARGET_LINUX_x86 -D TARGET_NO_AREC
+    CXXFLAGS += -fno-exceptions
+
+# Generic x86_64 (64 bit)
+else ifneq (,$(findstring AMD64,$(platform)))
+    X64_REC := 1
+    NOT_ARM := 1
+    CFLAGS += -D TARGET_LINUX_x86 -D TARGET_NO_AREC
+    CXXFLAGS += -fexceptions
+
+# Raspberry Pi 2
+else ifneq (,$(findstring rpi2,$(platform)))
+    MFLAGS += -marm -march=armv7-a -mtune=cortex-a9 -mfpu=neon -mfloat-abi=hard -funroll-loops
+    ASFLAGS += -march=armv7-a -mfpu=neon -mfloat-abi=softfp
+    CFLAGS += -D TARGET_BEAGLE -D TARGET_LINUX_ARMELv7 -DARM_HARDFP
+    INCS += -I/opt/vc/include/ -I/opt/vc/include/interface/vmcs_host/linux -I/opt/vc/include/interface/vcos/pthreads -I../linux-deps/include
+    LIBS += -L/opt/vc/lib/  -L../linux-deps/lib -lbcm_host
+    USE_GLES := 1
+
+# BeagleBone Black
+else ifneq (,$(findstring beagle,$(platform)))
+    CC_PREFIX ?= arm-none-linux-gnueabi-
+    MFLAGS += -marm -march=armv7-a -mtune=cortex-a9 -mfpu=neon -mfloat-abi=softfp -funroll-loops
+    ASFLAGS := -march=armv7-a -mfpu=neon -mfloat-abi=softfp
+    CFLAGS += -D TARGET_BEAGLE
+    USE_GLES := 1
+
+# Pandora
+else ifneq (,$(findstring pandora,$(platform)))
+    FOR_ARM := 1
```

Comment:
I guess I can remove this? `FOR_ARM` doesn't seem to be used anywhere except in the Pandora Makefile:
https://github.com/reicast/reicast-emulator/search?q=FOR_ARM